src/jshint.js

```diff
@@ -4584,7 +4605,25 @@ var JSHINT = (function() {
   FutureReservedWord("public", { es5: true, strictOnly: true });
   FutureReservedWord("short");
   FutureReservedWord("static", { es5: true, strictOnly: true });
-  FutureReservedWord("super", { es5: true });
+  FutureReservedWord("super", {
+    es5: true,
+    nud: function() {
+      if (state.inESNext() && checkPunctuators(state.tokens.next, "(")) {
```

Comment:
will this warn on this
```
// jshint esnext: false
function a() {
  super();
}
```
I noticed I don't get any warnings at all in 2.8.0 (e.g. super is reserved, super is not defined, super requires es6)

This would not be a regression so we can handle this seperately if you want? (unless I'm wrong of course)